`UpdateRoleMultiPartitionTest.shouldDistributeRoleUpdateCommand` is flaky

**Summary**

> Try to answer the following as best as possible

- How often does the test fail? so far, once [here](https://github.com/camunda/camunda/actions/runs/11438049438?pr=23754) 
- Does it block your work? no
- Do we suspect that it is a real failure? IDK

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
  shouldDistributeRoleUpdateCommand: io.camunda.zeebe.engine.processing.authorization.UpdateRoleMultiPartitionTest

Expecting actual to contain the specified subsequence but failed to find the element at subsequence index 0 in actual:
subsequence element not found in actual:
  (UPDATE, COMMAND, 1)
actual:
  [(CREATE, COMMAND, 1),
    (CREATED, EVENT, 1),
    (STARTED, EVENT, 1),
    (ENQUEUED, EVENT, 2),
    (DISTRIBUTING, EVENT, 2),
    (ENQUEUED, EVENT, 3),
    (DISTRIBUTING, EVENT, 3),
    (ACKNOWLEDGE, COMMAND, 2),
    (ACKNOWLEDGE, COMMAND, 3),
    (ACKNOWLEDGED, EVENT, 2),
    (ACKNOWLEDGED, EVENT, 3),
    (FINISH, COMMAND, 1),
    (FINISHED, EVENT, 1)]
subsequence:
  [(UPDATE, COMMAND, 1), (UPDATED, EVENT, 1), (STARTED, EVENT, 1)]
java.lang.AssertionError: 
Expecting actual to contain the specified subsequence but failed to find the element at subsequence index 0 in actual:
subsequence element not found in actual:
(UPDATE, COMMAND, 1)
actual:
[(CREATE, COMMAND, 1),
(CREATED, EVENT, 1),
(STARTED, EVENT, 1),
(ENQUEUED, EVENT, 2),
(DISTRIBUTING, EVENT, 2),
(ENQUEUED, EVENT, 3),
(DISTRIBUTING, EVENT, 3),
(ACKNOWLEDGE, COMMAND, 2),
(ACKNOWLEDGE, COMMAND, 3),
(ACKNOWLEDGED, EVENT, 2),
(ACKNOWLEDGED, EVENT, 3),
(FINISH, COMMAND, 1),
(FINISHED, EVENT, 1)]
subsequence:
[(UPDATE, COMMAND, 1), (UPDATED, EVENT, 1), (STARTED, EVENT, 1)]
at io.camunda.zeebe.engine.processing.authorization.UpdateRoleMultiPartitionTest.shouldDistributeRoleUpdateCommand(UpdateRoleMultiPartitionTest.java:63)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
</pre>
</details>

**Hypotheses**

> List any hypotheses if you have one; can be ommitted

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.


[LOGS](https://drive.google.com/file/d/1Um-HDhjrEB_EOvbxseNUmkgrLZr-R7-q/view?usp=drive_link)

--- a/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/AddEntityRoleMultiPartitionTest.java
+++ b/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/AddEntityRoleMultiPartitionTest.java
@@ -120,6 +120,9 @@ public class AddEntityRoleMultiPartitionTest {
   @Test
   public void distributionShouldNotOvertakeOtherCommandsInSameQueue() {
     // given the user creation distribution is intercepted
+    for (int partitionId = 2; partitionId <= PARTITION_COUNT; partitionId++) {
+      interceptUserCreateForPartition(partitionId);
+    }
     final var userKey =
         engine
             .user()
@@ -129,9 +132,6 @@ public class AddEntityRoleMultiPartitionTest {
             .withPassword("zabraboof")
             .create()
             .getKey();
-    for (int partitionId = 2; partitionId <= PARTITION_COUNT; partitionId++) {
-      interceptUserCreateForPartition(partitionId);
-    }
 
     // when
     final var name = UUID.randomUUID().toString();

--- a/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/AddPermissionAuthorizationMultiPartitionTest.java
+++ b/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/AddPermissionAuthorizationMultiPartitionTest.java
@@ -134,6 +134,10 @@ public class AddPermissionAuthorizationMultiPartitionTest {
   @Test
   public void distributionShouldNotOvertakeOtherCommandsInSameQueue() {
     // given the user creation distribution is intercepted
+    for (int partitionId = 2; partitionId <= PARTITION_COUNT; partitionId++) {
+      interceptUserCreateForPartition(partitionId);
+    }
+
     final var ownerKey =
         engine
             .user()
@@ -143,9 +147,6 @@ public class AddPermissionAuthorizationMultiPartitionTest {
             .withPassword("zabraboof")
             .create()
             .getKey();
-    for (int partitionId = 2; partitionId <= PARTITION_COUNT; partitionId++) {
-      interceptUserCreateForPartition(partitionId);
-    }
 
     // when
     engine

--- a/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/CreateRoleMultiPartitionTest.java
+++ b/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/CreateRoleMultiPartitionTest.java
@@ -99,6 +99,10 @@ public class CreateRoleMultiPartitionTest {
   @Test
   public void distributionShouldNotOvertakeOtherCommandsInSameQueue() {
     // given the user creation distribution is intercepted
+    for (int partitionId = 2; partitionId <= PARTITION_COUNT; partitionId++) {
+      interceptUserCreateForPartition(partitionId);
+    }
+
     engine
         .user()
         .newUser(UUID.randomUUID().toString())
@@ -106,10 +110,6 @@ public class CreateRoleMultiPartitionTest {
         .withPassword("baz")
         .withEmail("foobar@baz.com")
         .create();
-
-    for (int partitionId = 2; partitionId <= PARTITION_COUNT; partitionId++) {
-      interceptUserCreateForPartition(partitionId);
-    }
 
     // when
     final var name = UUID.randomUUID().toString();

--- a/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/RemoveEntityRoleMultiPartitionTest.java
+++ b/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/RemoveEntityRoleMultiPartitionTest.java
@@ -132,6 +132,9 @@ public class RemoveEntityRoleMultiPartitionTest {
   @Test
   public void distributionShouldNotOvertakeOtherCommandsInSameQueue() {
     // given the user creation distribution is intercepted
+    for (int partitionId = 2; partitionId <= PARTITION_COUNT; partitionId++) {
+      interceptUserCreateForPartition(partitionId);
+    }
     final var userKey =
         engine
             .user()
@@ -141,9 +144,6 @@ public class RemoveEntityRoleMultiPartitionTest {
             .withPassword("zabraboof")
             .create()
             .getKey();
-    for (int partitionId = 2; partitionId <= PARTITION_COUNT; partitionId++) {
-      interceptUserCreateForPartition(partitionId);
-    }
 
     // when
     final var name = UUID.randomUUID().toString();

--- a/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/UpdateRoleMultiPartitionTest.java
+++ b/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/UpdateRoleMultiPartitionTest.java
@@ -46,7 +46,8 @@ public class UpdateRoleMultiPartitionTest {
     assertThat(
             RecordingExporter.records()
                 .withPartitionId(1)
-                .limit(record -> record.getIntent().equals(CommandDistributionIntent.FINISHED)))
+                .limitByCount(
+                    record -> record.getIntent().equals(CommandDistributionIntent.FINISHED), 2))
         .extracting(
             io.camunda.zeebe.protocol.record.Record::getIntent,
             io.camunda.zeebe.protocol.record.Record::getRecordType,

--- a/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/user/DeleteUserMultiPartitionTest.java
+++ b/zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/user/DeleteUserMultiPartitionTest.java
@@ -135,6 +135,10 @@ public class DeleteUserMultiPartitionTest {
 
   @Test
   public void distributionShouldNotOvertakeOtherCommandsInSameQueue() {
+    // given the user creation distribution is intercepted
+    for (int partitionId = 2; partitionId <= PARTITION_COUNT; partitionId++) {
+      interceptUserCreateForPartition(partitionId);
+    }
     final var userRecord =
         engine
             .user()
@@ -143,11 +147,6 @@ public class DeleteUserMultiPartitionTest {
             .withEmail("foo@bar.com")
             .withPassword("password")
             .create();
-
-    // given the user creation distribution is intercepted
-    for (int partitionId = 2; partitionId <= PARTITION_COUNT; partitionId++) {
-      interceptUserCreateForPartition(partitionId);
-    }
 
     // when
     engine

